### PR TITLE
[PATCH API-NEXT v2] Update crypto tests

### PR DESCRIPTION
**Pull request by lumag**
- Print capabilities that were not covered by provided test vectors
- Fail the test if no test vectors were actually tested
- Add full-length HMAC test vectors in addition to truncated ones.

Last commit is only required till #352 is merged.

### CODE REVIEW

**Comment by lumag** [comment]
@Bill-Fischofer-Linaro @muvarov note, this PR is targeted api-next, but it does not contain any api-changes. If approved, this can be merged to master/tigermoth branches after -rc1.

**Comment by muvarov** [comment]
@lumag  that is correct. I do not expect more rebases of next.

File: test/validation/api/crypto/odp_crypto_test_inp.c [hunk 1]
```diff
@@ -628,32 +682,25 @@ static int check_alg_null(void)
 	return check_alg_support(ODP_CIPHER_ALG_NULL, ODP_AUTH_ALG_NULL);
 }
 
+#define ARRAY_SIZE(a) (sizeof(a) / sizeof(a[0]))
```
Comment (Bill-Fischofer-Linaro) [inline test/validation/api/crypto/odp_crypto_test_inp.c]: Checkpatch takes issue with this:
```
WARNING: Prefer ARRAY_SIZE(a)
#303: FILE: test/validation/api/crypto/odp_crypto_test_inp.c:671:
+#define ARRAY_SIZE(a) (sizeof(a) / sizeof(a[0]))
total: 0 errors, 1 warnings, 0 checks, 893 lines checked
NOTE: Ignored message types: AVOID_EXTERNS BIT_MACRO COMPARISON_TO_NULL DEPRECATED_VARIABLE NEW_TYPEDEFS PREFER_PRINTF PREFER_SCANF SPLIT_STRING SSCANF_TO_KSTRTO VOLATILE
0001-validation-crypto-print-untested-capabilities.patch has style problems, please review.
```
This seems to be another instance of a Linux kernel macro? False positive.

**Comment by lumag** [comment]
Replaced by #379

**Pull request closed without merging**